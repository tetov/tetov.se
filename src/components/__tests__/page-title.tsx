import React from "react";
import renderer from "react-test-renderer";
import PageTitle from "src/components/page-title";

describe("PageTitle", () => {
  it("PageTitle with custom prop and string child renders correctly", () => {
    const tree = renderer
      .create(
        <PageTitle itemProp="testline" mfClass="p-name">
          Title
        </PageTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PageTitle with default prop and string child renders correctly", () => {
    const tree = renderer.create(<PageTitle>Title</PageTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PageTitle with custom prop and JSX child renders correctly", () => {
    const tree = renderer
      .create(
        <PageTitle itemProp="JSX" mfClass="JSX">
          <a href="https://example.com/">Test</a>
        </PageTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PageTitle with default prop and JSX child renders correctly", () => {
    const tree = renderer
      .create(
        <PageTitle>
          <div className="header">Header</div>
        </PageTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PageTitle with articleHeader set to false renders a div", () => {
    const tree = renderer
      .create(
        <PageTitle articleHeader={false}>
          <div className="header">Header</div>
        </PageTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PageTitle with articleHeader set to true renders header", () => {
    const tree = renderer
      .create(
        <PageTitle articleHeader>
          <div className="header">Header</div>
        </PageTitle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
