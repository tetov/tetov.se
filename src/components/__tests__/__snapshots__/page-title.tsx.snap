// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle PageTitle with custom prop and JSX child renders correctly 1`] = `
<div
  className="mt-4 mb-8"
>
  <h1
    className="JSX text-4xl font-semibold leading-relaxed text-center"
    itemProp="JSX"
  >
    <a
      href="https://example.com/"
    >
      Test
    </a>
  </h1>
</div>
`;

exports[`PageTitle PageTitle with custom prop and string child renders correctly 1`] = `
<div
  className="mt-4 mb-8"
>
  <h1
    className="p-name text-4xl font-semibold leading-relaxed text-center"
    itemProp="testline"
  >
    Title
  </h1>
</div>
`;

exports[`PageTitle PageTitle with default prop and JSX child renders correctly 1`] = `
<div
  className="mt-4 mb-8"
>
  <h1
    className="p-name text-4xl font-semibold leading-relaxed text-center"
    itemProp="headline"
  >
    <div
      className="header"
    >
      Header
    </div>
  </h1>
</div>
`;

exports[`PageTitle PageTitle with default prop and string child renders correctly 1`] = `
<div
  className="mt-4 mb-8"
>
  <h1
    className="p-name text-4xl font-semibold leading-relaxed text-center"
    itemProp="headline"
  >
    Title
  </h1>
</div>
`;
